pcdsdevices/sequencer.py

```diff
@@ -0,0 +1,169 @@
+import time
+import logging
+from functools import partial
+
+from ophyd import Device, EpicsSignal, EpicsSignalRO, Component as Cpt
+from ophyd.status import DeviceStatus, SubscriptionStatus
+from ophyd.utils.epics_pvs import raise_if_disconnected
+
+
+logger = logging.getLogger(__name__)
+
+
+class EventSequencer(Device):
+    """
+    Event Sequencer
+
+    The LCLS Event Sequencer implemented as an Flyer; i.e it has the methods
+    :meth:`.kickoff`, :meth:`.complete` and :meth:`.collect`. This allows the
+    EventSequencer to be used succinctly with the `fly_during_wrapper` and
+    associated preprocessor.
+
+    Parameters
+    ----------
+    prefix: str
+        Base prefix of the EventSequencer
+
+    name : str
+        Name of Event Sequencer object
+
+    Examples
+    --------
+    Run the EventSequencer throughout my scan
+
+    .. code::
+
+        fly_during_wrapper(scan([det], motor, ...), [sequencer])
+
+    Run the EventSequencer at each step in my scan, after the completing the
+    motor move.
+
+    .. code::
+
+        def step_then_sequence(detectors, step, pos_cache)
+            yield from one_nd_step(detectors, step, pos_cache)
+            yield from kickoff(sequencer)
+            yield from complete(sequencer)  # Waits for sequence to complete if
+                                            # not in "Run Forever" mode
+
+        scan([det], motor, ..., per_step=step_then_sequence)
+    """
+    play_control = Cpt(EpicsSignal, ':PLYCTL')
+    sequence_length = Cpt(EpicsSignal, ':LEN')
+    current_step = Cpt(EpicsSignal, ':CURSTP')
+    play_count = Cpt(EpicsSignal, ':PLYCNT')
+    play_status = Cpt(EpicsSignalRO, ':PLSTAT', auto_monitor=True)
+    play_mode = Cpt(EpicsSignal, ':PLYMOD')
+
+    _default_read_attrs = ['play_status']
+    _default_configuration_attrs = ['play_mode', 'sequence_length']
+
+    def __init__(self, prefix, *, name=None,  **kwargs):
+        # Device initialization
+        super().__init__(prefix, name=name, **kwargs)
+        # Data caches
+        self._steps = list()
+        self._iterations = list()
+
+    @raise_if_disconnected
+    def kickoff(self):
```

Comment:
It seems like we're committing to having the `bluesky` flyer interface be the only way to start the sequencer. Just making sure there was a conscious decision to omit a `start` method similar to the `stop` method. I think this is ok but it needs to be stated. Would anyone ever be calling `seq.kickoff()` outside of a plan? I guess it's ok to force the `bluesky` API over another arbitrary API. End ramble.

Reply:
They are a little different. `stop` is used by `bluesky` to make sure that we don't leave the EventSequencer running if the scan exits in an ugly state.  `start` is a wrapper for user API. I am voting for forcing `bluesky` API

Reply:
I support your decision, but will direct all questions of "why is there a `stop` but no `start`" to you.